cocos/2d/CCProgressTimer.cpp

```diff
@@ -100,17 +100,17 @@ bool ProgressTimer::initWithSprite(Sprite* sp)
 
     CC_SAFE_RELEASE(_programState);
     CC_SAFE_RELEASE(_programState2);
-    initPipelineDescriptor(_customCommand, true);
-    initPipelineDescriptor(_customCommand2, false);
+    _programState = initPipelineDescriptor(_customCommand, true, _locMVP1, _locTex1); 
+    _programState2 = initPipelineDescriptor(_customCommand2, false, _locMVP2, _locTex2);
     
     return true;
 }
 
 ProgressTimer::~ProgressTimer(void)
 {
     CC_SAFE_RELEASE(_sprite);
-    CC_SAFE_RELEASE(_customCommand.getPipelineDescriptor().programState);
```

Comment:
`_customCommand.getPipelineDescriptor().programState` should only hold weak reference